### DISCUSSION TITLE
Updated Snipe IT and now getting a 502 Gateway error

### DISCUSSION THREAD

**Discussion by dagodad** [snipe/snipe-it#15002] [General]
I used the following 2 commands to upgrade Snipe-it 
git pull
php upgrade.php
In hopes to fix an import issue. 
before that we were getting an error that looked like it was being generated by Composer that composer needed PHP8.2. 
So that was upgraded from 7.4, and now we get the 502 gateway error. 
Now when I try to run a composer update I get this 
bcadmin@assetmanager:/var/www/snipe-it$ composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - Root composer.json requires PHP extension ext-curl * but it is missing from your system. Install or enable PHP's curl extension.
  Problem 2
    - eduardokum/laravel-mail-auto-embed is locked to version 2.11 and an update of this package was not requested.
    - eduardokum/laravel-mail-auto-embed 2.11 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
  Problem 3
    - rollbar/rollbar is locked to version v4.0.2 and an update of this package was not requested.
    - rollbar/rollbar v4.0.2 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
  Problem 4
    - spatie/laravel-backup is locked to version 8.8.1 and an update of this package was not requested.
    - spatie/laravel-backup 8.8.1 requires ext-zip ^1.14.0 -> it is missing from your system. Install or enable PHP's zip extension.
  Problem 5
    - spatie/laravel-ignition is locked to version 2.8.0 and an update of this package was not requested.
    - spatie/laravel-ignition 2.8.0 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
  Problem 6
    - tecnickcom/tc-lib-barcode is locked to version 1.18.4 and an update of this package was not requested.
    - tecnickcom/tc-lib-barcode 1.18.4 requires ext-bcmath * -> it is missing from your system. Install or enable PHP's bcmath extension.
  Problem 7
    - rollbar/rollbar v4.0.2 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
    - rollbar/rollbar-laravel v8.0.1 requires rollbar/rollbar ^4.0 -> satisfiable by rollbar/rollbar[v4.0.2].
    - rollbar/rollbar-laravel is locked to version v8.0.1 and an update of this package was not requested.

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/8.2/cli/php.ini
    - /etc/php/8.2/cli/conf.d/10-opcache.ini
    - /etc/php/8.2/cli/conf.d/10-pdo.ini
    - /etc/php/8.2/cli/conf.d/15-xml.ini
    - /etc/php/8.2/cli/conf.d/20-calendar.ini
    - /etc/php/8.2/cli/conf.d/20-ctype.ini
    - /etc/php/8.2/cli/conf.d/20-dom.ini
    - /etc/php/8.2/cli/conf.d/20-exif.ini
    - /etc/php/8.2/cli/conf.d/20-ffi.ini
    - /etc/php/8.2/cli/conf.d/20-fileinfo.ini
    - /etc/php/8.2/cli/conf.d/20-ftp.ini
    - /etc/php/8.2/cli/conf.d/20-gettext.ini
    - /etc/php/8.2/cli/conf.d/20-iconv.ini
    - /etc/php/8.2/cli/conf.d/20-phar.ini
    - /etc/php/8.2/cli/conf.d/20-posix.ini
    - /etc/php/8.2/cli/conf.d/20-readline.ini
    - /etc/php/8.2/cli/conf.d/20-shmop.ini
    - /etc/php/8.2/cli/conf.d/20-simplexml.ini
    - /etc/php/8.2/cli/conf.d/20-sockets.ini
    - /etc/php/8.2/cli/conf.d/20-sysvmsg.ini
    - /etc/php/8.2/cli/conf.d/20-sysvsem.ini
    - /etc/php/8.2/cli/conf.d/20-sysvshm.ini
    - /etc/php/8.2/cli/conf.d/20-tokenizer.ini
    - /etc/php/8.2/cli/conf.d/20-xmlreader.ini
    - /etc/php/8.2/cli/conf.d/20-xmlwriter.ini
    - /etc/php/8.2/cli/conf.d/20-xsl.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-curl --ignore-platform-req=ext-zip --ignore-platform-req=ext-bcmath` to temporarily ignore these required extensions.
Not sure if the system can be saved... Any help is appreciated. TIA

**Comment by mauro-miatello** [comment]
have you installed php8.2 and all modules? which os are you using?
